blog/content/post/2021-02-21-tigerlake-osf.md

```diff
@@ -0,0 +1,183 @@
+out what is wrong. Trying to narrow it down with debug FSP binary didn't help
+as well, because the FSP asserts in Thunderbolt/USB type C initialization.
+
+I am still not close to booting an operating system, but yet I have encountered
+many bugs in the meantime. This is the first time I have been put through so
+much effort to boot a reference platform. Previous platform I have tried
+(Cometlake-S or Kaby Lake U) did not cause almost any problem. The quality of
+the implementation of the Intel silicon support done by Intel employees
+themselves seem to have degraded. As a proof of that please red the following
+section about EDK2 MinPlatform.
+
+## EDK2
+
+EDK2 besides the open source common modules for UEFI compliant firmware it also
+contains the platform code hosted in a separate repository called
+[edk2-platforms](https://github.com/tianocore/edk2-platforms). This is where
+Intel publishes the reference board support code that integrates with FSP to
+boot reference platforms. The TigerLake open board packages have been published
+just few days ago. I gave it a try almost immediately with hope it will give
+better results.
+
+But on high hopes it ended, quite quickly. Although EDK2 is supposed to support
+GCC5, it occurred it is not always true. The freshly published code was not
+buildable with GCC5 when trying with the 3mdeb
+[edk2-docker](https://github.com/3mdeb/edk2-docker).
+
+When giving it a little thought it is not surprising. Visual Studio compiler is
+the one that dominates the ecosystem of firmware. Intel, AND and IBVs
+(Independent BIOS Vendors) use Windows and Microsoft compilers to build EDK2.
+Thus it was not tested with GCC compilers. The list of encountered problems is
+posted on EDK2 Bugzilla: https://bugzilla.tianocore.org/show_bug.cgi?id=3220
+
+Even though I have fixed all compilation issues for GCC5, I noticed that the
+packages contain support for LPDDR4 platform, again... The EDK2 FSP integration
+can work in two modes; FSP API and FSP dispatch mode. Switching between them is
+simply a build flag change. Gave a try to both without success. The FSP memory
+init return error, again... Changing the DDR memory signals routing as in
+coreboot did not help. Currently I am stuck at this problem which I have
+reported to the Bugzilla: https://bugzilla.tianocore.org/show_bug.cgi?id=3219
+
+## Possible solutions
+
+One of the obvious solutions that would be necessary is to add the build
+targets for TigerLake UP3 RVP DDR4 by Intel, hopefully validated build targets.
+To get rid of build issue under different toolchains a simple CI/CD would be
+more than enough to build test the reference platform with supported
+toolchains. coreboot already has the build testing of the patchsets sent for
+review with Jenkins for a long time. Moreover it is moving to validation on
+target hardware which EDK2 clearly lacks currently. It could be also great if
+Intel would publish validation results for FSP and open board packages. As far
```

Comment:
@miczyg1 please refer to Dasharo Transparent Validation System and recent blog post about testing

Reply:
Added references, please review